pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -43,31 +44,101 @@ const (
 	OVNKubeOverlayNetwork = "OVNKubernetesHybridOverlayNetwork"
 	// kubeProxyServiceName is the name of the kube-proxy Windows service
 	kubeProxyServiceName = "kube-proxy"
+	// remotePowerShellCmdPrefix holds the PowerShell prefix that needs to be prefixed  for every remote PowerShell
+	// command executed on the remote Windows VM
+	remotePowerShellCmdPrefix = "powershell.exe -NonInteractive -ExecutionPolicy Bypass "
 )
 
 var log = logf.Log.WithName("windows")
 
-// Windows is a wrapper for the WindowsVM interface
+// Windows contains all the  methods needed to configure a Windows VM to become a worker node
 type Windows interface {
-	types.WindowsVM
+	// ID returns the cloud provider ID of the VM
+	ID() string
+	// CopyFile copies the given file to the remote directory in the Windows VM. The remote directory is created if it
+	// does not exist
+	CopyFile(string, string) error
+	// Run executes the given command remotely on the Windows VM over a ssh connection and returns the combined output
+	// of stdout and stderr. If the bool is set, it implies that the cmd is to be execute in PowerShell. This function
+	// should be used in scenarios where you want to execute a command that runs in the background. In these cases we
+	// have observed that Run() returns before the command completes and as a result killing the process.
+	Run(string, bool) (string, error)
+	// Reinitialize re-initializes the Windows VM's SSH client
+	Reinitialize() error
 	Configure() error
 	ConfigureCNI(string) error
 	ConfigureHybridOverlay(string) error
 	ConfigureKubeProxy(string, string) error
-	Validate() error
 }
 
 // windows implements the Windows interface
 type windows struct {
-	types.WindowsVM
+	// id is the VM's cloud provider ID
+	id string
 	// workerIgnitionEndpoint is the Machine Config Server(MCS) endpoint from which we can download the
 	// the OpenShift worker ignition file.
 	workerIgnitionEndpoint string
+	// signer is used for authenticating against the VM
+	signer ssh.Signer
+	// conn is used to connect to and interact with the VM
+	conn *connectivity
 }
 
-// New returns a new instance of windows struct
-func New(vm types.WindowsVM, workerIgnitionEndpoint string) Windows {
-	return &windows{WindowsVM: vm, workerIgnitionEndpoint: workerIgnitionEndpoint}
+// New returns a new Windows instance constructed from the given WindowsVM
+// TODO: WINC-429 will replace WindowsVM with a MachineSet
+func New(vm types.WindowsVM, workerIgnitionEndpoint string, signer ssh.Signer) (Windows, error) {
+	// Validate the WindowsVM
+	if err := validate(vm); err != nil {
```

Comment:
Why to validate the windows vm here, we can do it the moment it got created before we call [this](https://github.com/openshift/windows-machine-config-operator/pull/77/commits/0f189ce63e7d36224be99929798a586777c7941e#diff-6a80c7c59a5c0c59d705c805a83830e3R80)

Reply:
I feel it is more natural to do this validation as part of the method that is taking an input (WNI VM today, `MachineSet` tomorrow) and returning a Windows WMCO object rather than outside it.